tests/test_diffraction_objects.py

```diff
@@ -350,21 +363,84 @@ def test_scale_to_bad(org_do_args, target_do_args, scale_inputs):
 
 
 @pytest.mark.parametrize(
-    "wavelength, xarray, yarray, xtype_1, xtype_2, value, expected_index",
+    "do_args, get_array_index_inputs, expected_index",
     [
-        # UC1: Exact match
-        (4 * np.pi, np.array([30.005, 60]), np.array([1, 2]), "tth", "tth", 30.005, [0]),
-        # UC2: Target value lies in the array, returns the (first) closest index
-        (4 * np.pi, np.array([30, 60]), np.array([1, 2]), "tth", "tth", 45, [0]),
-        (4 * np.pi, np.array([30, 60]), np.array([1, 2]), "tth", "q", 0.25, [0]),
-        # UC3: Target value out of the range, returns the closest index
-        (4 * np.pi, np.array([0.25, 0.5, 0.71]), np.array([1, 2, 3]), "q", "q", 0.1, [0]),
-        (4 * np.pi, np.array([30, 60]), np.array([1, 2]), "tth", "tth", 63, [1]),
+        # Test get_array_index() returns the expected index given xtype and value
+        # C1: Target value is in the xarray and xtype is identical, expect exact index match
```

Comment:
let's not modify this PR again, but I do wonder whether it would be more compact as
```
        (  # C1: Target value is in the xarray and xtype is identical, expect exact index match
            {
                "wavelength": 4 * np.pi,
                "xarray": np.array([30.005, 60]),
                "yarray": np.array([1, 2]),
                "xtype": "tth",
            },
            {
                "xtype": "tth",
                "value": 30.005,
            },
            [0],
        ),
```
than the current
```
    # C1: Target value is in the xarray and xtype is identical, expect exact index match
        (
            {
                "wavelength": 4 * np.pi,
                "xarray": np.array([30.005, 60]),
                "yarray": np.array([1, 2]),
                "xtype": "tth",
            },
            {
                "xtype": "tth",
                "value": 30.005,
            },
            [0],
        ),
```

Reply:
<img width="397" alt="Screenshot 2024-12-25 at 10 07 41 AM" src="https://github.com/user-attachments/assets/5349bb13-eebb-4f84-99f2-9621bd893696" />

@sbillinge yeah I like that! The automatic indentation also provides a good visual hierarchy!

Reply:
<img width="669" alt="Screenshot 2024-12-25 at 10 11 26 AM" src="https://github.com/user-attachments/assets/078ea4b0-b13b-4ac0-a9fb-a29ed3f826c8" />

For single-lined cases, I think we still want to maintain this format right?